catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventHandlerDelegate.java

```diff
@@ -0,0 +1,378 @@
+        this();
+        this.eventHandlers = Arrays.asList(eventHandlers);
+    }
+
+    public SaxEventHandlerDelegate(List<SaxEventHandler> eventHandlers) {
+        this();
+        this.eventHandlers = eventHandlers;
+    }
+
+    /**
+     * Takes in an {@link InputStream} returns a {@link Metacard},
+     * populated with all the {@link Attribute}s parsed by the {@link SaxEventHandlerDelegate#eventHandlers}
+     *
+     * @param inputStream an XML document that can be parsed into a Metacard
+     * @return a {@link Metacard},
+     * populated with all the {@link Attribute}s parsed by the {@link SaxEventHandlerDelegate#eventHandlers}
+     * @throws CatalogTransformerException
+     */
+    public Metacard read(InputStream inputStream) throws CatalogTransformerException {
+        configureEventHandlerLookup();
+
+        /*
+         * Create a new MetacardImpl with the proper MetacardType
+         */
+        Metacard metacard = new MetacardImpl(metacardType);
+
+        try {
+            InputSource newStream = new InputSource(new BufferedInputStream(inputStream));
+
+            /*
+             * Set the parser's ContentHandler to this delegate, which ensures the delegate receives all
+             * parse events that should be handled by a SaxEventHandler (startElement, endElement, characters, startPrefixMapping, etc)
+             * Set the parser's ErrorHandler to be a new InputTransformerHandler
+             */
+            parser.setContentHandler(this);
+            InputTransformerErrorHandler errorHandler = new InputTransformerErrorHandler();
+            parser.setErrorHandler(errorHandler);
+            parser.parse(newStream);
+
+            /*
+             * If any major errors occur during parsing, print them out
+             */
+            String errorsAndWarnings = errorHandler.getParseWarningsErrors();
+            if (!(errorsAndWarnings == null || errorsAndWarnings.isEmpty())) {
+                LOGGER.warn(errorHandler.getParseWarningsErrors());
+            }
+
+        } catch (IOException | SAXException e) {
+            LOGGER.debug("Exception thrown during parsing of inputStream", e);
+            throw new CatalogTransformerException("Could not properly parse metacard", e);
+        }
+
+        /*
+         * Populate metacard with all attributes constructed in SaxEventHandlers during parsing
+         */
+        for (SaxEventHandler eventHandler : eventHandlers) {
+            List<Attribute> attributes = eventHandler.getAttributes();
+            for (Attribute attribute : attributes) {
+                Attribute tmpAttr;
+
+                /*
+                 * If metacard already has values in the attribute, put them together into a multivalued list,
+                 * instead of simply overwriting the existing values.
+                 */
+                if ((tmpAttr = metacard.getAttribute(attribute.getName())) != null) {
```

Comment:
I'm pretty sure we can't change the `Metacard` interface unfortunately. Anyway, it seems to me that these would be doing the same thing, your way just prevents the need for a `null` check.